src/test/java/com/google/api/generator/engine/ast/JavaDocCommentTest.java

```diff
@@ -62,9 +62,42 @@ public void createJavaDocComment_sampleCode() {
         JavaDocComment.builder().addComment(comment).addSampleCode(sampleCode).build();
     String expected =
         "sample codes:\n"
-            + "<pre><code>\n"
+            + "<pre>{@code\n"
             + "resource = project/{project}/shelfId/{shelfId}\n"
-            + "</code></pre>";
+            + "}</pre>";
+    assertEquals(javaDocComment.comment(), expected);
+  }
+
+  @Test
+  public void createJavaDocComment_sampleCodeKeepIndent() {
+    String comment = "sample codes:";
+    String sampleCode = "try (condition = false) {\n" + "  int x = 3;\n" + "}";
+    JavaDocComment javaDocComment =
+        JavaDocComment.builder().addComment(comment).addSampleCode(sampleCode).build();
+    String expected =
+        "sample codes:\n"
+            + "<pre>{@code\n"
+            + "try (condition = false) {\n"
+            + "  int x = 3;\n"
+            + "}\n"
+            + "}</pre>";
+    assertEquals(javaDocComment.comment(), expected);
+  }
+
+  @Test
+  public void createJavaDocComment_sampleCodeKeepLineBreaker() {
+    String comment = "sample codes:";
+    String sampleCode =
+        "SubscriptionAdminSettings subscriptionAdminSettings =\n"
+            + "    SubscriptionAdminSettings.newBuilder().setEndpoint(myEndpoint).build();";
+    JavaDocComment javaDocComment =
+        JavaDocComment.builder().addComment(comment).addSampleCode(sampleCode).build();
+    String expected =
+        "sample codes:\n"
+            + "<pre>{@code\n"
+            + "SubscriptionAdminSettings subscriptionAdminSettings =\n"
+            + "    SubscriptionAdminSettings.newBuilder().setEndpoint(myEndpoint).build();\n"
+            + "}</pre>";
     assertEquals(javaDocComment.comment(), expected);
   }
```

Comment:
Consider merging the newlines and indents test (and perhaps adding some "bad" indents to demonstrate preservation), since this doesn't test our logic.

Reply:
I am trying to show the capacity of keeping the long line break in the @code tag. 
"SubscriptionAdminSettings subscriptionAdminSettings =\n"
            + "    SubscriptionAdminSettings.newBuilder().setEndpoint(myEndpoint).build();\n" is formatted string code.
<pre>{@code..}</pre> maintain the format.

Hope that I explain your concern. I am open to discuss on this.

Reply:
To rephrase the ask, can we combine the newlines and indents test, such that we can show the long line breaks, indent preservation, and also other "odd" indents into a single test?

Reply:
Add three cases to show `@code` preserve the format inside the block. Thanks for clarify with me.